File: library/src/main/java/org/incendo/interfaces/inventory/CachedInterfacesInventory.java
package org.incendo.interfaces.inventory;

import org.bukkit.inventory.ItemStack;

public abstract class CachedInterfacesInventory implements InterfacesInventory {

    @Override
    public final boolean set(final int row, final int column, final ItemStack item) {
        ItemStack current = this.get(row, column);

        if (current == item) {
            return false;
        }

        this.setInternal(row, column, item);
        return true;
    }

    protected abstract void setInternal(int row, int column, ItemStack item);

}

File: library/src/main/java/org/incendo/interfaces/inventory/CachedInterfacesInventory.kt
package org.incendo.interfaces.inventory

import org.bukkit.inventory.ItemStack

public abstract class CachedInterfacesInventory : InterfacesInventory {

    final override fun set(row: Int, column: Int, item: ItemStack?): Boolean {
        val current = get(row, column)

        if (current == item) {
            return false
        }

        setInternal(row, column, item)
        return true
    }

    protected abstract fun setInternal(row: Int, column: Int, item: ItemStack?)
}
